test/over_react_redux/connect_flux_test.dart

```diff
@@ -0,0 +1,776 @@
+          var dispatchButton = queryByTestId(bigCounter, 'button-increment');
+
+          click(dispatchButton);
+          click(smallDispatchButton);
+
+          await Future(() {});
+
+          // Big counter updated to 100
+          expect(findDomNode(bigCounter).innerHtml, contains('Count: 100'),
+              reason: 'Should have a count of 100');
+
+          // Normal counter incremented only 1 at both instances
+          expect(
+              findDomNode(queryByTestId(jacket.mountNode, 'outside')).innerHtml,
+              contains('Count: 1</div>'));
+          expect(findDomNode(bigCounter).innerHtml, contains('Count: 1</div>'));
+        });
+      });
+    });
+
+    group('properly handles parameter combinations for', () {
+      final testCases = {
+        'just mapStateToProps (case1)': {
+          'mapStateToProps': testMapStateToProps,
+        },
+        'just mapStateToPropsWithOwnProps (case2)': {
+          'mapStateToPropsWithOwnProps': testMapStateToPropsWithOwnProps,
+        },
+        'both mapStateToProps and mapActionsToProps (case3)': {
+          'mapStateToProps': testMapStateToProps,
+          'mapActionsToProps': testMapActionsToProps,
+        },
+        'just mapActionsToProps (case4)': {
+          'mapActionsToProps': testMapActionsToProps,
+        },
+        'both withOwnProps parameters (case5)': {
+          'mapStateToPropsWithOwnProps': testMapStateToPropsWithOwnProps,
+          'mapActionsToPropsWithOwnProps': testMapActionsToPropsWithOwnProps,
+        },
+        'just mapActionsToPropsWithOwnProps (case6)': {
+          'mapActionsToPropsWithOwnProps': testMapActionsToPropsWithOwnProps,
+        },
+        'mapStateToProps and mapActionsToPropsWithOwnProps (case7)': {
+          'mapStateToProps': testMapStateToProps,
+          'mapActionsToPropsWithOwnProps': testMapActionsToPropsWithOwnProps,
+        },
+        'mapStateToPropsWithOwnProps and mapActionsToProps (case8)': {
+          'mapStateToPropsWithOwnProps': testMapStateToPropsWithOwnProps,
+          'mapActionsToProps': testMapActionsToProps,
+        },
+      };
+
+      testParameterCases(testCases);
+    });
+
+    test('prints a warning when state is mutated directly', () async {
+      final ConnectedFluxComponent =
+          connectFlux<FluxStore, FluxActions, ConnectFluxCounterProps>(
+        mapStateToProps: (state) => (ConnectFluxCounter()
+          ..mutatedList = state.listYouDefShouldntMutate),
+        mapActionsToProps: (actions) => (ConnectFluxCounter()
+          ..mutateStoreDirectly = actions.mutateStoreDirectly),
+      )(ConnectFluxCounter);
+
+      final jacket = mount((ReduxProvider()..store = store1)(
+        (ConnectedFluxComponent()..addTestId('flux-component'))(),
+      ));
+
+      final fluxCounter = queryByTestId(jacket.mountNode, 'flux-component');
+      final fluxButton = queryByTestId(fluxCounter, 'button-increment');
+
+      final logs = await recordConsoleLogsAsync(() async {
```

Comment:
Awww yeah, `recordConsoleLogsAsync` came in handy! 😄

Reply:
Yeah it did! I might use it in the other ticket I'm working on now too.. it's super handy haha